inst/B: use the new instruction schema

Unique-Usman opened the pull request:
In (#686), a new instruction schema was created
for types/subtypes. This commit changes B/clmul,
B/clmulh, B/orn, B/rol, B/rolw, B/ror, B/rorw,
B/xnor instructions to use the new schema. They
all are of the same subtypes. It is also part of
effort in (#655).

dhower-qc commented:
I'm looking into this

dhower-qc commented:
I checked out your branch and ran `./do gen:instruction_appendix`. I did not see your error. Do you by chance have any files locally that weren't checked in?

ThinkOpenly commented:
> obj/database_obj.rb:240:in `source_line': No $source for misa (RuntimeError)

I see this occasionally, and am able to remedy it by:
```
./do clean
```

ThinkOpenly commented on spec/std/isa/inst/B/clmulh.yaml:
```diff
@@ -12,15 +12,17 @@ description: |
+    - inst_subtype/R/R-x.yaml#/data
+  opcodes:
+    funct7:
+      display_name: CLMULH
```
Per the spec, this should be "CLMUL".

ThinkOpenly replied:
This still needs attention.

ThinkOpenly replied:
```suggestion
      display_name: CLMUL
```

ThinkOpenly replied:
@Unique-Usman, look carefully at the image I posted above. It is for CLMULH, but the leftmost set of bits is labelled "MINMAX/CLMUL", not "CLMULH". Your changes have it as "CLMULH", when it should be "CLMUL" (or perhaps "MINMAX/CLMUL" to match the spec exactly).

Unique-Usman replied:
Ohh, thanks for the explanation. I really appreaciate it. I see that the display_name is something that both the func7 and func3 has and could be different. I think it is okay if we call it "CLMUL", the func3 will then be CLMULH. Thanks and sorry for the stress.

ThinkOpenly commented:
You need to rebase after #858 got merged, as it touches all of the files in `spec/std/isa/inst/B`. (Sorry about that!)

ThinkOpenly commented:
You need to fix the conflicts identified in the rebase:
```
+<<<<<<< HEAD
 encoding:
   match: 0110000----------101-----0110011
   variables:
@@ -21,6 +22,19 @@ encoding:
       location: 19-15
     - name: xd
       location: 11-7
+=======
+format:
+  $inherits:
+    - inst_subtype/R/R-x.yaml#/data
+  opcodes:
+    funct7:
+      display_name: ROR
+      value: 0b0110000
+    funct3:
+      display_name: ROR
+      value: 0b101
+    opcode: { $inherits: inst_opcode/OP.yaml#/data }
+>>>>>>> ae38542c (inst/B: use the new instruction schema)
 access:
```

All of the conflicts are flagged with the same tags:
```
<<<<<<< HEAD
(code from the new "base" for the rebase)
=======
(your code)
>>>>>>> (your commit information)
```

You'll need to edit those files that have conflicts to fix them up. Let me know if you need help.

Unique-Usman commented:
> You need to fix the conflicts identified in the rebase:
> 
> ```
> +<<<<<<< HEAD
>  encoding:
>    match: 0110000----------101-----0110011
>    variables:
> @@ -21,6 +22,19 @@ encoding:
>        location: 19-15
>      - name: xd
>        location: 11-7
> +=======
> +format:
> +  $inherits:
> +    - inst_subtype/R/R-x.yaml#/data
> +  opcodes:
> +    funct7:
> +      display_name: ROR
> +      value: 0b0110000
> +    funct3:
> +      display_name: ROR
> +      value: 0b101
> +    opcode: { $inherits: inst_opcode/OP.yaml#/data }
> +>>>>>>> ae38542c (inst/B: use the new instruction schema)
>  access:
> ```
> 
> All of the conflicts are flagged with the same tags:
> 
> ```
> <<<<<<< HEAD
> (code from the new "base" for the rebase)
> =======
> (your code)
> >>>>>>> (your commit information)
> ```
> 
> You'll need to edit those files that have conflicts to fix them up. Let me know if you need help.

Thanks, I mistakenly left out the two files when I was doing the rebase.

ThinkOpenly commented:
> I already added a comment to what @ThinkOpenly said and I have not gotten feedback on it.

@Unique-Usman , what comment do you need feedback?

FYA, I have several comments that need your action/feedback, related to changing "CLMULH" to "CLMUL" in one place.

Unique-Usman commented:
@ThinkOpenly, I added the comment below your feedback. Can you see it ?

Unique-Usman commented:
> > @ThinkOpenly, I added the comment below your feedback. Can you see it ?
> 
> Maybe I can't? I don't see any comments from you that are questions for which there was no response. Maybe repost? (If they're marked "pending", then they did not get submitted.)

actually everything was saying pending, I have added lot of comment which was not seen. I already submitted it, it should be visible now.
